cg/meta/archive/ddn_dataflow.py

```diff
@@ -0,0 +1,222 @@
+from cg.constants.constants import APIMethods, FileFormat
+from cg.exc import DdnDataflowAuthenticationError
+from cg.io.controller import APIRequest, ReadStream
+from cg.models.cg_config import DDNDataFlowConfig
+
+OSTYPE: str = "Unix/MacOS"
+ROOT_TO_TRIM: str = "/home"
+
+
+class DataflowEndpoints(str, Enum):
+    """Enum containing all DDN dataflow endpoints used."""
+
+    GET_AUTH_TOKEN = "auth/token"
+    REFRESH_AUTH_TOKEN = "auth/token/refresh"
+    ARCHIVE_FILES = "files/archive"
+    RETRIEVE_FILES = "files/retrieve"
+
+
+class ResponseFields(str, Enum):
+    """Enum containing all DDN dataflow endpoints used."""
+
+    REFRESH = "refresh"
+    ACCESS = "access"
+    EXPIRE = "expire"
+    RETRIEVE_FILES = "files/retrieve"
+
+
+class TransferData(BaseModel):
+    """Model for representing a singular object transfer."""
+
+    source: str
+    destination: str
+    _metadata = None
+
+    def correct_source_root(self):
+        """Trims the source path from its root directory."""
+        self.source = f"/{Path(self.source).relative_to(ROOT_TO_TRIM).as_posix()}"
+
+    def correct_destination_root(self):
+        """Trims the destination path from its root directory."""
+        self.destination = f"/{Path(self.destination).relative_to(ROOT_TO_TRIM).as_posix()}"
+
+    def add_repositories(self, source_prefix: str, destination_prefix: str):
+        """Prepends the given repositories to the source and destination paths."""
+        self.source = source_prefix + self.source
+        self.destination = destination_prefix + self.destination
+
+
+class TransferPayload(BaseModel):
+    """Model for representing a Dataflow transfer task."""
+
+    files_to_transfer: List[TransferData]
+    osType: str = OSTYPE
+    createFolder: bool = False
+
+    def correct_source_root(self):
+        """Trims the source path from its root directory for all objects in the transfer."""
+        for transfer_data in self.files_to_transfer:
+            transfer_data.correct_source_root()
+
+    def correct_destination_root(self):
+        """Trims the destination path from its root directory for all objects in the transfer."""
+        for transfer_data in self.files_to_transfer:
+            transfer_data.correct_destination_root()
+
+    def add_repositories(self, source_prefix: str, destination_prefix: str):
+        """Prepends the given repositories to the source and destination paths all objects in the
+        transfer."""
+        for transfer_data in self.files_to_transfer:
+            transfer_data.add_repositories(
+                source_prefix=source_prefix, destination_prefix=destination_prefix
+            )
+
+    def dict(self, **kwargs) -> dict:
+        """Creates a correctly structured dict to be used as the request payload."""
+        payload: dict = super().dict(exclude={"files_to_transfer"})
+        payload["pathInfo"] = [transfer_data.dict() for transfer_data in self.files_to_transfer]
+        payload["metadataList"] = []
```

Comment:
```suggestion
        payload["metadata"]: List = []
```

Reply:
Same here regarding the naming of the fields